aws-sdk-core/lib/aws-sdk-core/plugins/s3_request_signer.rb

```diff
@@ -6,28 +6,15 @@ module Plugins
     # @api private
     class S3RequestSigner < Seahorse::Client::Plugin
 
-      class SigningHandler < RequestSigner::Handler
+      option :signature_version, 'v4'
 
-        # List of regions that support older S3 signature versions.
-        # All new regions only support signature version 4.
-        V2_REGIONS = Set.new(%w(
-          us-east-1
-          us-west-1
-          us-west-2
-          ap-northeast-1
-          ap-southeast-1
-          ap-southeast-2
-          sa-east-1
-          eu-west-1
-          us-gov-west-1
-        ))
+      class SigningHandler < RequestSigner::Handler
 
         def call(context)
           require_credentials(context)
-          version = signature_version(context)
-          case version
-          when /v4/ then apply_v4_signature(context)
-          when /s3/ then apply_v2_signature(context)
+          case context.config.signature_version
+          when 'v4' then apply_v4_signature(context)
+          when 's3' then apply_s3_legacy_signature(context)
           else raise "unsupported signature version #{version.inspect}"
```

Comment:
This is a good opportunity to improve this existing error message. Add into the message the valid signature versions.